### ISSUE TITLE
HTTP requests made by app extremely slow

### ISSUE THREAD

**Issue by alXDraghiciu** [dotnet/runtime#79405]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Hi.

Suddenly for the last few days, all my ASP.NET projects started behaving strangely **on the local machine**.

All the HTTP requests they make to 3rd parties seem to be extremely slow. In addition, they can't seem to process parallel requests from the client anymore (like they only use one thread).

Either using RESTSharp or HttpClient, it always take forever.

Simply fetching the OAuth config from Microsoft Identity authentication (v2.0/.well-known/openid-configuration) takes about 3 minutes.

While other requests take similarly between 3 to 5 minutes.

I have no idea what has started all this, other than I had an electricity blackout some days ago. And I also updated Visual Studio which might have updated other packages? I have no idea.

**But no other software is affected but ASP/.NET.**

Everything works fine when deployed, localhost is the only affected environment.

The only solution I've found so far is increasing timeout. I tried flushing DNS but no effect.

If anyone knows what could be happening or has any suggestions, I would appreciate.

### .NET Version

7.0.100

### Anything else?

Microsoft Visual Studio Community 2022 (64-bit) - Current
Version 17.4.2

.NET SDK:
 Version:   7.0.100
 Commit:    e12b7af219

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19044
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.100\

Host:
  Version:      7.0.0
  Architecture: x64
  Commit:       d099f075e4

.NET SDKs installed:
  5.0.400 [C:\Program Files\dotnet\sdk]
  6.0.102 [C:\Program Files\dotnet\sdk]
  6.0.200 [C:\Program Files\dotnet\sdk]
  6.0.300 [C:\Program Files\dotnet\sdk]
  7.0.100 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  arm64 [C:\Program Files\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\arm64\InstallLocation]
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

**Comment by wfurt** [comment]
Besides what @Tratcher  already recommended I would suggest two do packet capture with Wireshark .hat should show what is going on on the wire. Do you see same issue when trying to access sites using IPAddress?

**Comment by alXDraghiciu** [comment]
> I've seen machine proxy settings interfere like this. https://learn.microsoft.com/en-us/dotnet/fundamentals/networking/http/httpclient#http-proxy
> 
> Also security software on the machine can cause issues like this. E.g. Norton, Mcafee, etc.

I only have Windows Defender installed. I've disabled it but no effect.
Machine config only has this line on the proxy. Doesn't look enabled:
```
<section name="defaultProxy" type="System.Net.Configuration.DefaultProxySection, System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=bXXXX" />
```

**Comment by alXDraghiciu** [comment]
> 

Hi. How would I go about accessing sites using IP address? Replacing the 3rd party URL's my app is calling with IP addresses in the code?

Or trying to access localhost by IP address.

**Comment by alXDraghiciu** [comment]
I've installed Wireshark but I'm not sure what to look for. Any suggestions?

**Comment by wfurt** [comment]
You would also see if there are any attempts to connect proxy or if the connection goes out directly to target server. Another tool you can try is Fiddler. It installs itself as proxy and it allows you to monitor and analyze HTTP traffic.

**Comment by alXDraghiciu** [comment]
> 

I've installed Fiddler. 

Initially I was using Visual Studio Debugger and noticed that it acted like a proxy for the API. I closed it and ran in command line. Now the app itself was making the requests, but they were still slow.

I made a test project that requested data from https://jsonplaceholder and it worked fast.
I then added Azure AD Authorization to the test app as well and it slowed to a halt.
login.windows.net is extremely slow, and also Zoho People API.

But strangely, if I request a token using OAuth Postman option, it works fast and without problems. Just the API's seem to have speed issues.
